macos/DefenseClawMac/Tests/DependencyLockValidatorTests.py

```
@@ -0,0 +1,102 @@
+    def validate(self, lock: str, source_url: str = SCANNER_URL) -> subprocess.CompletedProcess[str]:
+        with tempfile.TemporaryDirectory() as temporary:
+            root = Path(temporary)
+            pyproject = root / "pyproject.toml"
+            lock_path = root / "requirements.lock"
+            pyproject.write_text(
+                '[project]\nname = "fixture"\nversion = "1.0.0"\n'
+                f'dependencies = ["cisco-ai-skill-scanner @ {source_url}"]\n',
+                encoding="utf-8",
+            )
+            lock_path.write_text(lock, encoding="utf-8")
+            return subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)